src/app.js

```diff
@@ -0,0 +1,133 @@
+require("express");
+require("express-async-errors");
+
+const path = require("path");
+const session = require("express-session");
+const AWS = require("aws-sdk");
+const DynamoDBStore = require("connect-dynamodb")(session);
+const wizard = require('hmpo-form-wizard');
+const logger = require("hmpo-logger")
+
+const commonExpress = require("di-ipv-cri-common-express");
+
+const setHeaders = commonExpress.lib.headers;
+const setScenarioHeaders = commonExpress.lib.scenarioHeaders;
+const setAxiosDefaults = commonExpress.lib.axios;
+
+const { setAPIConfig, setOAuthPaths } = require("./lib/settings");
+const { setI18n } = require("di-ipv-cri-common-express/src/lib/i18next");
+const steps = require("./app/bav/steps");
+const fields = require("./app/bav/fields");
+
+const {
+  API,
+  APP,
+  PORT,
+  SESSION_SECRET,
+  SESSION_TABLE_NAME,
+  SESSION_TTL,
+} = require("./lib/config");
+
+const { setup } = require("hmpo-app");
+
+const loggerConfig = {
+  console: true,
+  consoleLevel: 'debug',
+  consoleJSON: true,
+  app: false,
+};
+
+AWS.config.update({
+  region: "eu-west-2",
+});
+const dynamodb = new AWS.DynamoDB();
+
+const dynamoDBSessionStore = new DynamoDBStore({
+  client: dynamodb,
+  table: SESSION_TABLE_NAME,
+});
+
+const sessionConfig = {
+  cookieName: "service_session",
+  secret: SESSION_SECRET,
+  cookieOptions: { maxAge: SESSION_TTL },
+  ...(SESSION_TABLE_NAME && { sessionStore: dynamoDBSessionStore }),
+};
+
+const helmetConfig = require("di-ipv-cri-common-express/src/lib/helmet");
+
+const { app, router } = setup({
+  config: { APP_ROOT: __dirname },
+  port: PORT,
+  logs: loggerConfig,
+  session: sessionConfig,
+  helmet: helmetConfig,
+  redis: SESSION_TABLE_NAME ? false : commonExpress.lib.redis(),
+  urls: {
+    public: "/public",
+  },
+  publicDirs: ["../dist/public"],
+  views: [
+    path.resolve(
+      path.dirname(require.resolve("di-ipv-cri-common-express")),
+      "components"
+    ),
+    "views",
+  ],
+  translation: {
+    allowedLangs: ["en"],
+    fallbackLang: ["en"],
+    cookie: { name: "lng" },
+  },
+  middlewareSetupFn: (app) => {
+    app.use(setHeaders);
+  },
+  dev: true,
+});
+
+setI18n({
+  router,
+  config: {
+    secure: true,
+    cookieDomain: APP.ANALYTICS.DOMAIN,
+  },
+});
+
+app.get("nunjucks").addGlobal("getContext", function () {
+  return {
+    keys: Object.keys(this.ctx),
+    ctx: this.ctx.ctx,
+  };
+});
+
+setAPIConfig({
+  app,
+  baseUrl: API.BASE_URL,
+  sessionPath: API.PATHS.SESSION
+});
+
+setOAuthPaths({ app, entryPointPath: APP.PATHS.BAV });
+
+router.use(setScenarioHeaders);
+router.use(setAxiosDefaults);
+
+router.use("/oauth2", commonExpress.routes.oauth2);
+
+const wizardOptions = {
+  name: "cri-bav-front",
+  journeyName: "bav",
+  templatePath: "bav",
+};
+
+router.use(wizard(steps, fields, wizardOptions));
+
+// router.use((err, req, res, next) => {
```

Comment:
How come this is commented out?

Reply:
Yeah let's uncomment this please @Stelios-CB happy to chat about what it does but tl;dr it catches when users don't have redirectURI which results in a server_error to IPV (which we don't want)

Reply:
Was WIP - fixed now